src/cmap/auth/gssapi.ts

```diff
@@ -21,65 +31,159 @@ export class GSSAPI extends AuthProvider {
       return callback(Kerberos['kModuleError']);
     }
 
-    const username = credentials.username;
-    const password = credentials.password;
-    const mechanismProperties = credentials.mechanismProperties;
-    const gssapiServiceName =
+    const { username, password, mechanismProperties } = credentials;
+    const serviceName =
       mechanismProperties['gssapiservicename'] ||
       mechanismProperties['gssapiServiceName'] ||
       'mongodb';
+    const canonicalizeHostName =
```

Comment:
I know you didn't write this code, but I think there's more opportunity to clean it up here. `canonicalizeHostName` is only used within `performGssapiCanonicalizeHostName`, so it looks like we can just do:
```
performGssapiCanonicalizeHostName(host, mechanismProperties, (err, host) => {
  ...
}); 
```